samples/cpp/videowriter.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+  @file videowriter.cpp
+  @brief A sample for VideoWriter and VideoCapture with options to specify video codec, fps and resolution
+  @date April 05, 2024
+*/
+
+
+#include <opencv2/core.hpp>
+#include <opencv2/videoio.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+#include <iostream>
+#include <map>
+
+using namespace cv;
+using namespace std;
+
+inline map<string, int> fourccByCodec() {
+    map<string, int> res;
+    res["h264"] = VideoWriter::fourcc('H', '2', '6', '4');
+    res["h265"] = VideoWriter::fourcc('H', 'E', 'V', 'C');
+    res["mpeg2"] = VideoWriter::fourcc('M', 'P', 'E', 'G');
+    res["mpeg4"] = VideoWriter::fourcc('M', 'P', '4', '2');
+    res["mjpeg"] = VideoWriter::fourcc('M', 'J', 'P', 'G');
+    res["vp8"] = VideoWriter::fourcc('V', 'P', '8', '0');
+    return res;
+}
+
+inline map<string, Size> sizeByResolution() {
+    map<string, Size> res;
+    res["720p"] = Size(1280, 720);
+    res["1080p"] = Size(1920, 1080);
+    res["4k"] = Size(3840, 2160);
+    return res;
+}
+
+int main(int argc, char** argv) {
+    const String keys =
+        "{help h usage ? |      | Print help message   }"
+        "{fps           |30    | fix frame per second for encoding (supported: fps > 0)   }"
+        "{codec         |mjpeg | codec name (supported: 'h264', 'h265', 'mpeg2', 'mpeg4', 'mjpeg', 'vp8')  }"
```

Comment:
The codec list depends on supported back-ends. Mjpeg is implemented by OpenCV itself. I propose to rewrite:
"Supported codecs depend on OpenCV Configuration. Try one of 'h264', 'h265', 'mpeg2', 'mpeg4', 'mjpeg', 'vp8'"